src/ruleset.rs

```diff
@@ -491,14 +491,32 @@ impl RulesetCreated {
         }
     }
 
+    /// Creates a copy of this object and duplicates the file descriptor with
+    /// CLOEXEC set.
+    ///
+    /// On error, returns a wrapped [`RulesetError::Duplicate`].
+    pub fn duplicate(&self) -> Result<RulesetCreated, RulesetError> {
+        let dup_fd = unsafe { libc::fcntl(self.fd, libc::F_DUPFD_CLOEXEC) };
+        if dup_fd == -1 {
+            return Err(RulesetError::Duplicate(Error::last_os_error()));
+        }
+
+        Ok(RulesetCreated {
+            fd: dup_fd,
+            no_new_privs: self.no_new_privs,
+            requested_handled_fs: self.requested_handled_fs,
+            compat: self.compat,
+        })
+    }
+
     /// Attempts to restrict the calling thread with the ruleset
     /// according to the best-effort configuration
     /// (see [`RulesetCreated::set_best_effort()`]).
     /// Call `prctl(2)` with the `PR_SET_NO_NEW_PRIVS`
     /// according to the ruleset configuration.
     ///
     /// On error, returns a wrapped [`RestrictSelfError`].
-    pub fn restrict_self(mut self) -> Result<RestrictionStatus, RulesetError> {
+    pub fn restrict_self(&mut self) -> Result<RestrictionStatus, RulesetError> {
```

Comment:
I need to ponder this change a bit. This designed was initially chosen to avoid unexpected behavior, but this use case could change it.